File: 1203/app/src/main/java/com/example/a1203/MainActivity.java
package com.example.a1203;

import android.content.Intent;
import android.database.Cursor;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private EditText editTextName, editTextField1, editTextField2;
    private CatDatabaseHelper dbHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        dbHelper = new CatDatabaseHelper(this);

        editTextName = findViewById(R.id.editTextName);
        editTextField1 = findViewById(R.id.editTextField1);
        editTextField2 = findViewById(R.id.editTextField2);
        Button buttonAdd = findViewById(R.id.buttonAdd);
        Button buttonView = findViewById(R.id.buttonView);
        Button buttonRename = findViewById(R.id.buttonRename);
        Button buttonDelete = findViewById(R.id.buttonDelete);

        buttonAdd.setOnClickListener(view -> {
            String name = editTextName.getText().toString();
            String field1 = editTextField1.getText().toString();
            String field2 = editTextField2.getText().toString();
            if (!name.isEmpty()) {
                dbHelper.addCat(name, field1, field2);
                Toast.makeText(this, "Добавлена кошечка: " + name, Toast.LENGTH_SHORT).show();
            }
        });

        buttonView.setOnClickListener(view -> {
            Intent intent = new Intent(MainActivity.this, CatListActivity.class);
            startActivity(intent);
        });

        buttonRename.setOnClickListener(view -> {
            dbHelper.renameLastCat();
            Toast.makeText(this, "Последняя кошечка переименована в 'Сахарок'", Toast.LENGTH_SHORT).show();
        });

        buttonDelete.setOnClickListener(view -> {
            dbHelper.deleteAllAndAddDefaultCats();
            Toast.makeText(this, "Удалены все записи. Добавлены 5 новых кошечек.", Toast.LENGTH_SHORT).show();
        });
    }
}

File: 28044thlabonkotlin/app/src/main/java/com/example/a2804_4th_lab_on_kotlin/MainActivity.kt
package com.example.a2804_4th_lab_on_kotlin

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var editTextName: EditText
    private lateinit var editTextField1: EditText
    private lateinit var editTextField2: EditText
    private lateinit var dbHelper: CatDatabaseHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        dbHelper = CatDatabaseHelper(this)

        editTextName = findViewById(R.id.editTextName)
        editTextField1 = findViewById(R.id.editTextField1)
        editTextField2 = findViewById(R.id.editTextField2)
        val buttonAdd = findViewById<Button>(R.id.buttonAdd)
        val buttonView = findViewById<Button>(R.id.buttonView)
        val buttonRename = findViewById<Button>(R.id.buttonRename)
        val buttonDelete = findViewById<Button>(R.id.buttonDelete)

        buttonAdd.setOnClickListener {
            val name = editTextName.text.toString()
            val field1 = editTextField1.text.toString()
            val field2 = editTextField2.text.toString()

            if (name.isNotEmpty()) {
                dbHelper.addCat(name, field1, field2)
                Toast.makeText(this, "Добавлена кошечка: $name", Toast.LENGTH_SHORT).show()
            }
        }

        buttonView.setOnClickListener {
            startActivity(Intent(this, CatListActivity::class.java))
        }

        buttonRename.setOnClickListener {
            dbHelper.renameLastCat()
            Toast.makeText(this, "Последняя кошечка переименована в 'Сахарок'", Toast.LENGTH_SHORT).show()
        }

        buttonDelete.setOnClickListener {
            dbHelper.deleteAllAndAddDefaultCats()
            Toast.makeText(this, "Удалены все записи. Добавлены 5 новых кошечек.", Toast.LENGTH_SHORT).show()
        }
    }
}